mesheryctl/internal/cli/root/system/token_test.go

```diff
@@ -0,0 +1,426 @@
+package system
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+)
+
+func TestTokenCreateCmd(t *testing.T) {
+	// get current directory
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatal("Not able to get current working directory")
+	}
+	currDir := filepath.Dir(filename)
+	tests := []utils.CmdTestInput{
+		{
+			Name:                 "create the passed token with default location",
+			Args:                 []string{"token", "create", "Default2"},
+			ExpectedResponse:     "create_default.golden",
+			ExpectedResponseYaml: "create_default.yaml",
+			ExpectErr:            false,
+		},
+		{
+			Name:                 "create the passed token with passed location",
+			Args:                 []string{"token", "create", "Default2", "-f", "~/.meshery/auth.json"},
+			ExpectedResponse:     "create.golden",
+			ExpectedResponseYaml: "create.yaml",
+			ExpectErr:            false,
+		},
+		{
+			Name:                 "creating a token which already exists",
+			Args:                 []string{"token", "create", "Default"},
+			ExpectedResponse:     "create_err.golden",
+			ExpectedResponseYaml: "create_err.yaml",
+			ExpectErr:            true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			utils.SetupCustomContextEnv(t, currDir+"/testdata/token/"+tt.ExpectedResponseYaml)
+			var b *bytes.Buffer
+			var e *bytes.Buffer
+			if tt.ExpectErr {
+				e = utils.SetupLogrusGrabTesting(t)
+				SystemCmd.SetErr(e)
+			} else {
+				b = utils.SetupLogrusGrabTesting(t)
+				SystemCmd.SetOut(b)
+			}
+			SystemCmd.SetArgs(tt.Args)
+			var actualResponse string
+			err := SystemCmd.Execute()
+			if err != nil {
+				if !tt.ExpectErr {
+					t.Error(err)
+					return
+				}
+			}
+			//Check the stdout/stderr against the golden file
+			if tt.ExpectErr {
+				actualResponse = e.String()
+			} else {
+				actualResponse = b.String()
+			}
```

Comment:
Can you look at perf/apply_test.go Line 240? I think we can write this whole ExpectError thing in a better format, so that file might help you in a way.

Reply:
Hey, @Revolyssup did you checked the file? Can we refactor this code? TBH I believe that we can.

Reply:
There we do not have the buffer for errors. I wanted to have a buffer for that too, for debugging purposes.

Reply:
why multiple buffer-s?

Reply:
one for errstream, one for output stream from command. Errstream helps in debugging.